external_samples/component.py

```diff
@@ -71,6 +71,14 @@ def get_url(self) -> str:
     def get_version(self) -> tuple[int, int, int]:
         pass
 
+    @abstractmethod
```

Comment:
I think these should be default methods instead of abstract so that components that don't need them don't have to implement them.